buggy_script.py

```
@@ -0,0 +1,48 @@
+    def greet(self):
+        # 4. 类型错误：尝试将字符串和整数直接连接
+        print("Hello, I am " + self.name + " and I am " + self.age + " years old.")
+
+def main():
+    print("Welcome to the shop!")
+    
+    # 5. 逻辑错误：运算符误用 (^ 在 Python 中是异或，不是幂运算)
+    square_area = 10 ^ 2 
+    print(f"Area calculation check: {square_area}") 
+
+    # 6. 语法错误：在 if 条件中使用了赋值运算符 (=) 而不是比较运算符 (==)
```

Ruff (0.14.10)

Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)